Add post-mortem post about events table id overflow (#360)
openSUSE/obs-landing: _posts/deployments/2022-05-19-post-mortem.md

---
layout: post
title: "Post-Mortem: Events Table Overflow on May 18, 2022"
category: deployments
author: Saray
---

There was a severe service degradation of our [reference server](https://build.opensuse.org). On 2022-05-18 the events system of OBS started to fail, causing problems in different areas of the application. We want to give you some insight into what happened.

## Impact

> OBS users started to have issues (500 errors) when working on different areas of the application.

Reviewer: Are you referring to http status 500, right? I would use "http status 500", or not mentioning that at all and just say "internal errors"

Author: Ok, l removed it.

> The team decided to switch the application to maintenance mode to avoid new incoming requests leading to issues while working on the solution.

Reviewer: I would say that we were losing data, and we switched to maintenance to stop losing more

Author: I rephrased most of the paragraphs in the Impact section. Please have another look at them.

> build.opensuse.org was offline for 16 minutes. No one was able to work with the API or user interface during that time.

Author: Added

There was some data loss as well, some events were not stored in the database. Therefore, some emails were not sent and OBS did not report back some results for SCM/CI integrations.

## Root Causes

The events table reached the maximum value possible for the id column. It overflowed and it was not possible to store new events on the table.

## Trigger

The number of events increased fast lately since we introduced the relationship events.

## Detection

> OBS users started to report 500 errors via IRC channel.

Reviewer 2: ```suggestion
OBS users started to report internal server errors via IRC channel.
```

Reviewer 2: or mention its http code 500, otherwise it sound like the amount was 500

Author: done

A colleague reported the increase of errors via Slack.
Thousands of errors were tracked by the error tracking service.

## Resolution

We switched the application to maintenance mode while working on the solution.

> Then, we created and ran a migration to change the id column type from integer to bigint to accept such a huge number of events.

Reviewer 2: ```suggestion
We created and ran a database migration to change the id column type from integer to bigint in order to store a bigger amount of events.
```

Author: done

We finally restarted the server.

## Action Items

- Disabling the relationship events subscriptions by default.
- Adding to the code base the migration we ran on the fly.
- Improving monitoring and alerts to be notified about these kinds of issues earlier.

## Lessons Learned

### What Went Well

Collaboration among the team to resolve this.

> Communication from other colleagues and users to inform use about the issue.

Reviewer 2: ```suggestion
Communication from other colleagues and users to inform us about the issue.
```

Author: Sorry, I didn't see this comment before. Now it's pushed.

---
layout: post
title: "Post-Mortem: Events Table Overflow on May 18, 2022"
category: deployments
author: Saray
---

There was a severe service degradation of our [reference server](https://build.opensuse.org). On 2022-05-18 the events system of OBS started to fail, causing problems in different areas of the application. We want to give you some insight into what happened.

## Impact

OBS users started to have issues when working on different areas of the application.

The issue was causing some data loss, so the team decided to switch the application to maintenance mode to stop loosing more data while working on the solution.

> build.opensuse.org was offline for 16 minutes. No one was able to work with the API or user interface during that time. Other services depending on OBS (like software.opensuse.org) where taken down by this as well.

Reviewer: ```suggestion
https://build.opensuse.org was offline for 16 minutes. No one was able to work with the API or user interface during that time. Other services depending on OBS (like https://software.opensuse.org) where taken down by this as well.
```
